### DISCUSSION TITLE
Raspberry Pi 5 + Ubuntu 24.04 LTS

### DISCUSSION THREAD

**Discussion by SvenVD** [OpenSurv/OpenSurv#3] [Hardware + Operating system]
All experiences and tests with OpenSurv on Raspberry Pi 5 and Ubuntu 24.04 LTS
( old thread [here ](https://github.com/SvenVD/rpisurv/issues/194))

For Raspberry Pi 5 + RPi OS Bookworm look here: https://github.com/OpenSurv/OpenSurv/discussions/3

**Comment by aniceberg** [comment]
After following the steps from the rpisurv thread, I was able to install OpenSurv on an RPi 5 8GB running Ubuntu 24.04.1 LTS  and had the demo video playing fine. However once I edited the monitor1.yml and restarted lightdm, I got this error:
```
pi@rpi5:~/OpenSurv$ sudo service lightdm status
● lightdm.service - Light Display Manager
     Loaded: loaded (/usr/lib/systemd/system/lightdm.service; indirect; preset: enabled)
     Active: active (running) since Mon 2024-11-25 20:55:15 EST; 12min ago
       Docs: man:lightdm(1)
    Process: 9259 ExecStartPre=/bin/sh -c [ "$(basename $(cat /etc/X11/default-display-manager 2>/dev/null))" = "lightdm" ] (code=exited, status=0/SUCCESS)
   Main PID: 9263 (lightdm)
      Tasks: 9 (limit: 9063)
     Memory: 48.6M (peak: 76.9M)
        CPU: 1.817s
     CGroup: /system.slice/lightdm.service
             ├─9263 /usr/sbin/lightdm
             ├─9652 /usr/lib/xorg/Xorg -core :0 -seat seat0 -auth /var/run/lightdm/root/:0 -nolisten tcp vt7 -novtswitch
             └─9804 lightdm --session-child 12 20

Nov 25 20:55:15 rpi5 systemd[1]: Starting lightdm.service - Light Display Manager...
Nov 25 20:55:15 rpi5 systemd[1]: Started lightdm.service - Light Display Manager.
Nov 25 20:55:16 rpi5 lightdm[9282]: pam_unix(lightdm-autologin:session): session opened for user opensurv(uid=1003) by (uid=0)
Nov 25 20:55:19 rpi5 lightdm[9660]: pam_unix(lightdm-greeter:session): session opened for user lightdm(uid=123) by (uid=0)
Nov 25 20:55:19 rpi5 lightdm[9660]: gkr-pam: couldn't unlock the login keyring.
Nov 25 20:55:19 rpi5 lightdm[9804]: pam_succeed_if(lightdm:auth): requirement "user ingroup nopasswdlogin" not met by user "opensurv"
```
So basically the RPi is stuck at the Ubuntu login page-- it shows the "pi" user account I created initially, the opensurv user, and a default Ubuntu login.

Any tips to get this working?

**Reply by SvenVD** [reply]
> After following the steps from the rpisurv thread, I was able to install OpenSurv on an RPi 5 8GB running Ubuntu 24.04.1 LTS and had the demo video playing fine. However once I edited the monitor1.yml and restarted lightdm, I got this error:
> ...
> So basically the RPi is stuck at the Ubuntu login page-- it shows the "pi" user account I created initially, the opensurv user, and a default Ubuntu login.
> 
> Any tips to get this working?

You probably made a typo in monitor1.yml what is in /home/opensurv/logs/main.log?

**Comment by aniceberg** [comment]
Open further comparison, I realized that rpisurv's "camera_streams" was renamed to "streams" in opensurv, so my monitor1.yml copypasta caused an issue with ScreenManager. Once I fixed that, it worked perfectly on my RPi v5.

If you're curious, you can see the attached main.log file [main.log](https://github.com/user-attachments/files/17950761/main.log)

Thanks!

**Reply by SvenVD** [reply]
Great to hear it is working. Thanks for reporting back! Can you share 

- how much memory your rpi 5 has
- how many streams you are displaying and what codec and resolution the source streams are configured with

**Reply by aniceberg** [reply]
> Great to hear it is working. Thanks for reporting back! Can you share
> 
> * how much memory your rpi 5 has

8GB

> * how many streams you are displaying and what codec and resolution the source streams are configured with

3 streams total:
- 2 * 960x720, h.265 (Unifi cameras using "enhanced encoding")
- 1 * 1920 x 1080 h.265 (main profile), 20fps, 3000kbps CBR

Load averages hover between high 1.0 to low 2.0s, each core reporting low 20% to low 30% usage.

**Comment by skowron0** [comment]
Hi, I install OpenSurv on RPi 5 8GB with Pi OS (Debian GNU/Linux 12 (Bookworm)). When start i see only default lightdm desktop login screen.
log:
```
2024/12/11 11:01:36 - l_default - INFO - Starting opensurv 1.2
2024/12/11 11:01:37 - DAEMON:
2024/12/11 11:01:37 - DAEMON: (xfwm4:1285): xfwm4-WARNING **: 11:01:37.800: Another compositing manager is running on screen 0
2024/12/11 11:01:37 - DAEMON:
2024/12/11 11:01:37 - DAEMON: (xfwm4:1285): xfwm4-WARNING **: 11:01:37.808: Failed to connect to session manager: Failed to connect to the session manager: SESSION_MANAGER environment variable not defined
2024/12/11 11:01:38 - DAEMON: /usr/bin/unclutter: someone created a sub-window to my sub-window! giving up
2024/12/11 11:42:41 - l_default - INFO - Starting opensurv 1.2
2024/12/11 11:42:42 - DAEMON:
2024/12/11 11:42:42 - DAEMON: (xfwm4:14175): xfwm4-WARNING **: 11:42:42.985: Failed to connect to session manager: Failed to connect to the session manager: SESSION_MANAGER environment variable not defined
2024/12/11 11:42:43 - DAEMON: /usr/bin/unclutter: someone created a sub-window to my sub-window! giving up

```
Whats is wrong?

**Reply by skowron0** [reply]
Ok, when I reboot again works ok, but I must manualy write command:
`systemctl restart lightdm.service` to start.
It is possible start automatical?

**Reply by SvenVD** [reply]
It should start automatically, I do not know what went wrong in your case but you can do `systemctl enable lightdm.service` to have it autostarted

**Reply by skowron0** [reply]
Unfortunately, that didn't help. Right after starting, I see a login screen to the console. When I log in via ssh and manually type `systemctl restart lightdm.service` it works very well with four streams. Maybe I can send you a log? I'd be happy to test opensurv, I've been using your earlier versions of raspisurv for years, they're great!

**Reply by SvenVD** [reply]
I created discussion Raspberry Pi 5 + RPi OS Bookworm look here: https://github.com/OpenSurv/OpenSurv/discussions/21 so we can leave this discussion for Ubuntu

Can you post /home/opensurv/logs/main.log  logs there

**Comment by diekgait** [comment]
Tested with some MP4 streams. These parameters give the best/stable playback:
freeform_advanced_mpv_options: "--rtsp-transport=tcp --profile=low-latency --untimed  --gpu-dumb-mode=yes --opengl-glfinish=yes --hwdec=drm-copy"

**Reply by s-kaz2285** [reply]
hi @diekgait where can I setup this settings?

**Reply by diekgait** [reply]
> hi @diekgait where can I setup this settings?

In your monitor.yml config file:
```
essentials:
  screens:
    - camera_streams:
        - url: "rtsp://streamurl"
          freeform_advanced_mpv_options: "--rtsp-transport=tcp --profile=low-latency --untimed --gpu-dumb-mode=yes --opengl-glfinish=yes --hwdec=drm-copy"
```

**Reply by s-kaz2285** [reply]
> > hi @diekgait where can I setup this settings?
> 
> In your monitor.yml config file:
> 
> ```
> essentials:
>   screens:
>     - camera_streams:
>         - url: "rtsp://streamurl"
>           freeform_advanced_mpv_options: "--rtsp-transport=tcp --profile=low-latency --untimed --gpu-dumb-mode=yes --opengl-glfinish=yes --hwdec=drm-copy"
> ```

Thanks

**Comment by aniceberg** [comment]
Did you notice a difference whether those streams were encoded in h.264 or
h.265?

**Reply by aniceberg** [reply]
After testing these parameters, my lone h.264 stream became blocky and less sharp. Removing the parameters restored the image fidelity.

My h.265 streams seem okay using it, but I haven't noticed an improvements of either quality or lag.

**Comment by HuskyGen** [comment]
Hello,  
I am currently running RPI 5 with '2024-11-19-raspios-bookworm-arm64' image.  I am able to get Opensurv working.  I only have one RTSP Stream.  I would like to enable audio for this one stream but am not sure how to do it.  
I did set the parameter 'enableaudio: True' in my monitor1.yml file.  But no audio. 
I am using HDMI cable out to a monitor with speakers built in.  In RaspiOS, I do get audio if I were to load a video from the web or the same RTSP stream via VLC player. 
Any help would be great.  
My current monitor1.yml file:

```c++
essentials:
   screens:
      - streams:
           -url: "rtsp://user:pass@xxx.xxx.xxx.xxx/cam/realmonitor?channel=1&subtype=0"
             freeform_advanced_mpv_options: "--rtsp-transport=tcp --profile=low-latency --untimed --gpu-dumb-mode=yes --opengl-glfinish=yes --hwdec=drm-copy"
             enableaudio: True

```

**Reply by aniceberg** [reply]
Your config lines appears to have extra spaces. Remove the single space before both "freeform" and "enableaudio: True" lines so that they are inline with the "u" of the "url" line above it like this:

```
    - streams:
        - url: "rtsp://user:pass@xxx.xxx.xxx.xxx/cam/realmonitor?channel=1&subtype=0"
          freeform_advanced_mpv_options: "--rtsp-transport=tcp --profile=low-latency --untimed --gpu-dumb-mode=yes --opengl-glfinish=yes --hwdec=dr>
          enableaudio: true
```

**Comment by HuskyGen** [comment]
Hello Aniceberg,

Thank you for your help.  
I went ahead and modified my spacing to match yours,  however still no audio.
```c#
essentials:
   screens:
      - streams:
           - url: "rtsp://user:pass@xxx.xxx.xxx.xxx/cam/realmonitor?channel=1&subtype=0"
             freeform_advanced_mpv_options: "--rtsp-transport=tcp --profile=low-latency --untimed --gpu-dumb-mode=yes --opengl-glfinish=yes --hwdec=dr>
             enableaudio: true
```

**Comment by HuskyGen** [comment]
Actually I am mistaken,  I do hear audio coming from the RTSP steam.  However it is extremely quiet,  I did not notice it at first.  The speakers on the  monitor are turned up to 100% but it seems like the audio volume on the Raspberry Pi 5 is turned down.  Is there a way to turn the volume up on the RPi 5?

**Comment by HuskyGen** [comment]
Just in case someone runs into the same issue,  I was able to amplify the audio from the Raspberry Pi 5 by running the following command in Terminal: 'amixer cset numid=3 50%' where numid=3 is the Master Playback volume(might be different in your setup but you can run 'amixer contents' to see which numid is your Master Playback volume, and #% is the volume level you can set.   
My setup uses a touchscreen monitor with build in speakers and is connected to the RPI 5 through HDMI.

**Comment by HuskyGen** [comment]
I have another question.  If later on I decide to add a second camera and have it display on the same monitor as 50/50 split left and right,  does Opensurv support,  since I have a touchscreen monitor,  double tapping on one of the camera streams to make it full screen  and then if I were to double tap it again,  it will shrink back to displaying both streams 50/50 split screen again?

**Reply by aniceberg** [reply]
Opensurv doesn't have that capability, though you might be able to get it to work by setting up additional streams for each of the two cameras that will only display that selected camera. Your primary stream would be accessed by pressing "0" or F1 on your keyboard, the second stream with "1" or F2, your third stream with "2" or F3, and so and so forth. You then need to use a program that can take screen input from a specific region and convert that into a Keyboard command that matches the stream you want to bring up. When Opensurv receives the simulated keyboard input, it can then launch the appropriate stream.

You can then have it revert automatically after some idle time period by sending 0 or F1, or perhaps you set aside a smaller region on one of the screen corners that will send the keyboard command that opens the primary stream.

**Comment by SvenVD** [comment]
https://github.com/OpenSurv/OpenSurv/discussions/68

**Comment by HoBu5** [comment]
i have a Pi-5 8GB Ubunto 24.04 LTS
Installation was ok
OpenSurv demo is ok
stop/restart lightdm.service is ok
But if i add a rtsp://username:pwd@ip:554 command to monitor1.yml the monitor start ubunto-destop not lightdm. I didn't found a way to get the stream from any camera.
logfile-info:
2025/10/21 12:20:11 - DAEMON:
2025/10/21 12:20:11 - DAEMON: (xfwm4:42583): xfwm4-WARNING **: 12:20:11.477: Failed to connect to session manager: Failed to connect to the session manager: SESSION_MANAGER environment variable not defined
2025/10/21 12:20:11 - l_default - INFO - Starting opensurv 1.3
2025/10/21 12:22:56 - DAEMON:
2025/10/21 12:22:56 - DAEMON: (xfwm4:43541): xfwm4-WARNING **: 12:22:56.222: Failed to connect to session manager: Failed to connect to the session manager: SESSION_MANAGER environment variable not defined
2025/10/21 12:22:56 - l_default - INFO - Starting opensurv 1.3
2025/10/21 12:22:56 - DAEMON: Traceback (most recent call last):
2025/10/21 12:22:56 - DAEMON: File "/home/opensurv/lib/surveillance.py", line 96, in <module>
2025/10/21 12:22:56 - DAEMON: screen_manager=ScreenManager(f'screen_manager_{count}', monitor, enable_caching_next_screen, disable_pygame)
2025/10/21 12:22:56 - DAEMON: ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025/10/21 12:22:56 - DAEMON: File "/home/opensurv/lib/core/ScreenManager.py", line 28, in __init__
2025/10/21 12:22:56 - DAEMON: self._init_screens()
2025/10/21 12:22:56 - DAEMON: File "/home/opensurv/lib/core/ScreenManager.py", line 201, in _init_screens
2025/10/21 12:22:56 - DAEMON: self.all_screens.append(Screen(str(self.name) + "_screen" + str(counter), screen_cfg, self.monitor, self.background_drawinstance))
2025/10/21 12:22:56 - DAEMON: ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025/10/21 12:22:56 - DAEMON: File "/home/opensurv/lib/core/Screen.py", line 30, in __init__
2025/10/21 12:22:56 - DAEMON: self.streams_cfg = screen_cfg["streams"]
2025/10/21 12:22:56 - DAEMON: ~~~~~~~~~~^^^^^^^^^^^
2025/10/21 12:22:56 - DAEMON: KeyError: 'streams'
2025/10/21 12:22:57 - DAEMON: X connection to :0 broken (explicit kill or server shutdown).

i also tried:
mpv rtsp://user:pwd@ip:554
[ffmpeg] tcp: Port missing in uri
[lavf] avformat_open_input() failed
Failed to recognize file format.
Exiting... (Errors when loading file)

has anyone some info for me ???
btw i also ordered:  GMKtec G5 Mini-PC N97 only for this project.
I need OpenSurv running.

**Reply by HoBu5** [reply]
info:
pi@pi-desktop:~$ mpv
mpv 0.37.0 Copyright © 2000-2023 mpv/MPlayer/mplayer2 projects
libplacebo version: v6.338.2
FFmpeg version: 6.1.1-3ubuntu5
FFmpeg library versions:
   libavutil       58.29.100
   libavcodec      60.31.102
   libavformat     60.16.100
   libswscale      7.5.100
   libavfilter     9.12.100
   libswresample   4.12.100

**Reply by HoBu5** [reply]
Error found !
Password of camera has a character "#" included.
That was the problem. Removing the "#" in the password and everything is fine.

**Comment by linek1980** [comment]
I have a huge increase performance in PI5 : OS Raspberry Pi OS, Debian Trixie. 
freeform_advanced_mpv_options: "--rtsp-transport=udp --profile=low-latency --untimed --opengl-glfinish=yes --gpu-dumb-mode=no --hwdec=drm-copy --vo=gpu"

If your camera has eelay / lag, use additional --vd-lavc-threads=1 if the delay is still there change vo=gpu to --vo=sdl
